Add support for OKC settings

gangli02 opened the pull request:
- Added "wifi config" command.
- Add OKC parameter to "wifi config" command.
   Set okc to 1 to enable opportunistic key caching.

gangli02 committed:
modules: hostap: add support for OKC settings

- Added support for OKC settings

Signed-off-by: Gang Li <gang.li_1@nxp.com>

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -1054,6 +1062,16 @@ struct wifi_ap_config_params {
+struct wifi_config_params {
+	/** Parameter used to identify the different STA parameters */
+	enum wifi_config_param type;
+#if defined(CONFIG_WIFI_NM_WPA_SUPPLICANT)
```
I understand, but wpa_supplicant has many such configs/knobs, so, was wondering if it would make sense to introduce a `wifi_nm_shell` or expose `wpa_cli` directly and if a command is generic then we can add it to Wi-Fi shell.

krish2718 commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -4287,6 +4353,18 @@ SHELL_SUBCMD_ADD((wifi), ps_exit_strategy, NULL,
 
+SHELL_SUBCMD_ADD((wifi), config, NULL,
+		 "Configure STA parameters.\n"
+#if defined(CONFIG_WIFI_NM_WPA_SUPPLICANT)
+		 "Please refer to wpa_supplicant.conf to set the following options,\n"
+		 "============ IEEE 802.11 related configuration ============\n"
+		 "-o, --okc=<0/1>: Opportunistic Key Caching. 0: disable, 1: enable\n"
+		 "===========================================================\n"
+#endif
```
as we had just discussed that this is generic command, may be remove supplicant define and also fix the help? You can always return ENOTSUP for embedded supp case

gangli02 replied:
Sure, let me remove it

krish2718 commented on modules/hostap/src/supp_api.c:
```diff
@@ -2336,3 +2336,31 @@ int supplicant_dpp_dispatch(const struct device *dev, struct wifi_dpp_params *pa
+			wpa_printf(MSG_ERROR, "Failed to set OKC");
+			goto out;
+		}
+		wpa_printf(MSG_INFO, "Set OKC: %d", params->okc);
```
```suggestion
		wpa_printf(MSG_DEBUG, "Set OKC: %d", params->okc);
```

gangli02 committed:
net: l2: wifi: add wifi config command

- Added "wifi config" command.
- Add OKC parameter to "wifi config" command.
  Set okc to 1 to enable opportunistic key caching.

Signed-off-by: Gang Li <gang.li_1@nxp.com>
